resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -257,8 +254,36 @@ private[spark] class KubernetesClusterSchedulerBackend(
     }
     true
   }
+
+  override def createDriverEndpoint(properties: Seq[(String, String)]): DriverEndpoint = {
+    new KubernetesDriverEndpoint(rpcEnv, properties)
+  }
+
+  /**
+   * Override the DriverEndpoint to add extra logic for the case when
+   * an executor is disconnected.
+   */
+  private class KubernetesDriverEndpoint(rpcEnv: RpcEnv, sparkProperties: Seq[(String, String)])
+    extends DriverEndpoint(rpcEnv, sparkProperties) {
+
+    /**
+     * We assume for now that we should create a replacement executor pod when it is lost.
+     * TODO: if spark scales down the number of executors (in dynamic allocation mode),
```

Comment:
We can check that an executor is individually healthy with a liveness probe on it.  But another failure mode is when it gets network partitioned from the driver.  It might create a lot of traffic on the driver if every probe check on the executor also verified connectivity to the driver with a ping.